### PR TITLE
Mapped a few goals

### PR DESCRIPTION
**Pull request by liach**

### CODE REVIEW

**Review by Prospector — APPROVED** [review]

**Review by SquidDev — APPROVED** [review]

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping [hunk 1]
```diff
@@ -1,4 +1,8 @@
 CLASS aqm net/minecraft/entity/mob/ZombieEntity
+	CLASS aqm$a DestroyEggGoal
+		METHOD a onDestory (Lbbp;Let;)V
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: onDestory?

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
+		METHOD a onStepProgress (Lbbq;Let;)V
+		METHOD i getDesiredSquareDistanceToTarget ()D
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: SquaredDistance

File: mappings/net/minecraft/entity/passive/IronGolemEntity.mapping [hunk 3]
```diff
@@ -1,5 +1,6 @@
 CLASS amh net/minecraft/entity/passive/IronGolemEntity
 	FIELD a IRON_GOLEM_FLAGS Lpr;
+	FIELD b findVillageDelay I
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/passive/IronGolemEntity.mapping:3]: villageSearchTicks? Or what does this do?

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/passive/IronGolemEntity.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/entity/passive/IronGolemEntity.mapping:3]: ticks to wait before searching for and joining a village

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: when the goal destroys a block

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: Fixed.

File: mappings/net/minecraft/entity/mob/ZombieEntity.mapping (hunk 1 above)
Comment (asiekierka) [reply mappings/net/minecraft/entity/mob/ZombieEntity.mapping]: Typo!

**Pull request merged**